platform/stm32f4xx/export/platform/i2c_bus.hpp

```diff
@@ -0,0 +1,648 @@
+#ifndef PLATFORM_I2C_BUS_HPP_
+#define PLATFORM_I2C_BUS_HPP_
+
+#include <stm32f4xx_i2c.h>
+#include <stm32f4xx_rcc.h>
+
+#include <platform/irq_manager.hpp>
+
+#include <common/i2c.hpp>
+
+#include <sys/types.h>
+
+namespace ecl {
+
+enum class i2c_mode {
+    POLL,
+    IRQ
+};
+
+template< i2c_device        dev,
+          i2c_mode          mode,
+          uint32_t          clock_speed,
+          uint16_t          operation_mode,
+          uint16_t          duty_cycle,
+          uint16_t          own_address1,
```

Comment:
That is name convention in SPL, so it matches on I2C_OwnAddress1, but yes, I agree that it does not make much sense. Will fix it. Thanks